packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2841 @@
+            showDecoration: (controller._anchor!._parent?._orientation ?? Axis.horizontal) == Axis.vertical,
+            child: child ?? const SizedBox(),
+          ),
+        );
+      },
+      menuChildren: widget.menuChildren,
+      child: widget.child,
+    );
+  }
+
+  void _handleFocusChange() {
+    if (_buttonFocusNode.hasPrimaryFocus) {
+      if (!_menuController.isOpen) {
+        _menuController.open();
+      }
+    } else {
+      if (!_menuController._anchor!._menuScopeNode.hasFocus && _menuController.isOpen) {
+        _menuController.close();
+      }
+    }
+  }
+}
+
+/// The type of builder function used by [MenuAnchor.builder] to build the
+/// widget that the [MenuAnchor] surrounds.
+///
+/// The `context` is the context that the widget is being built in.
+///
+/// The `controller` is the [MenuController] that can be used to open and close
+/// the menu with.
+///
+/// The `child` is an optional child supplied as the [MenuAnchor.child]
+/// attribute. The child is intended to be incorporated in the result of the
+/// function.
+typedef MenuAnchorChildBuilder = Widget Function(
+  BuildContext context,
+  MenuController controller,
+  Widget? child,
+);
+
+/// A widget used to mark the "anchor" for a set of submenus, defining the
+/// region used to position the menu, which can be done either with an explicit
+/// location, or with an alignment.
+///
+/// When creating a menu with [MenuBar] or a [SubmenuButton], a [MenuAnchor] is
+/// not needed, since they provide their own internally.
+///
+/// The [MenuAnchor] is meant to be a slightly lower level interface than
+/// [MenuBar], used in situations where a [MenuBar] isn't appropriate, or to
+/// construct widgets or screen regions that have submenus.
+///
+/// {@tool dartpad}
+/// This example shows how to use a [MenuAnchor] to wrap a button and open a
+/// cascading menu from the button.
+///
+/// ** See code in examples/api/lib/material/menu_bar/menu_anchor.0.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to use a [MenuAnchor] to create a cascading context
+/// menu in a region of the view, positioned where the user clicks the mouse
+/// with Ctrl pressed.
+///
+/// ** See code in examples/api/lib/material/menu_bar/menu_anchor.1.dart **
+/// {@end-tool}
+class MenuAnchor extends StatefulWidget with DiagnosticableTreeMixin {
+  /// Creates a const [MenuAnchor].
+  ///
+  /// The [menuChildren] argument is required.
+  const MenuAnchor({
+    super.key,
+    this.controller,
+    this.focusNode,
+    this.style,
+    this.onOpen,
+    this.onClose,
+    this.alignmentOffset = Offset.zero,
+    this.clipBehavior = Clip.none,
+    required this.menuChildren,
+    this.builder,
+    this.child,
+  })  : _isBar = false,
+        _isPositioned = false;
+
+  /// Creates a const [MenuAnchor] that can be positioned instead of aligned to
+  /// the region.
+  ///
+  /// Using this constructor also turns off the [TapRegion] mechanism present
```

Comment:
This feels like it should just be a general config option: Should tapping on the region occupied by the MenuAnchor close the menu or not (tapping outside the region will always close it)?